src/mesh/parallel/fci.cxx

```diff
@@ -94,9 +97,43 @@ FCIMap::FCIMap(Mesh &mesh, int dir, bool zperiodic)
 
   // Add the boundary region to the mesh's vector of parallel boundaries
   mesh.addBoundaryPar(boundary);
+  
+  // Cell corners
+  Field3D xt_prime_corner(&mesh), zt_prime_corner(&mesh);
+  xt_prime_corner.allocate();
+  zt_prime_corner.allocate();
 
-  interp->calcWeights(xt_prime, zt_prime);
+  for (int x = mesh.xstart; x <= mesh.xend; x++) {
+    for (int y = mesh.ystart; y <= mesh.yend; y++) {
+      for (int z = 0; z < mesh.LocalNz - 1; z++) {
```

Comment:
Could this be range-based loop over `RGN_ALL`?

Reply:
Possibly a similar issue for the `z+1` indexing as RGN_NOBNDRY visits all z currently?

Reply:
Good point, I guess it would need a custom region! Probably better like this in that case. It's in setup, so not likely to be a pinch point requiring omp